### PR TITLE
[14.0][ADD] product_search_by_display_name

### PR DESCRIPTION
**Pull request by cuongnmtm**

### CODE REVIEW

**Review by hieulucky111 — APPROVED** [review]
Code review
LGTM

**Review by CoraVanga — APPROVED** [review]
LGTM

**Review by Leonidas-VII — APPROVED** [review]
LGTM

File: product_search_by_display_name/models/product_product.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    _inherit = "product.product"
+
+    display_name = fields.Char(
+        automatic=True,
+        compute="_compute_display_name",
+        search="_search_display_name",
+    )
+
+    def _search_display_name(self, operator, value):
+        recs = self.search([]).filtered_domain([("display_name", operator, value)])
```
Comment (DucTruongKomit) [inline product_search_by_display_name/models/product_product.py]:
```suggestion
        recs = self.search([("display_name", operator, value)])
```

File: product_search_by_display_name/models/product_template.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    _inherit = "product.template"
+
+    display_name = fields.Char(
+        automatic=True,
+        compute="_compute_display_name",
+        search="_search_display_name",
+    )
+
+    def _search_display_name(self, operator, value):
+        recs = self.search([]).filtered_domain([("display_name", operator, value)])
```
Comment (DucTruongKomit) [inline product_search_by_display_name/models/product_template.py]:
```suggestion
        recs = self.search([("display_name", operator, value)])
```

Comment (cuongnmtm) [reply product_search_by_display_name/models/product_template.py]: The field display_name is not stored so it is not searchable.

File: product_search_by_display_name/models/product_product.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    display_name = fields.Char(
+        automatic=True,
```
Comment (hildickethan) [inline product_search_by_display_name/models/product_product.py:11]: i think automatic=True is already applied to all display_name fields

File: product_search_by_display_name/models/product_product.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        compute="_compute_display_name",
+        search="_search_display_name",
+    )
+
+    def _search_display_name(self, operator, value):
+        recs = self.search([]).filtered_domain([("display_name", operator, value)])
+        return [("id", "in", [r.id for r in recs or []])]
```
Comment (hildickethan) [inline product_search_by_display_name/models/product_product.py]:
```suggestion
        return [("id", "in", recs.ids)]
```

File: product_search_by_display_name/models/product_product.py (hunk 3 above)
Comment (cuongnmtm) [reply product_search_by_display_name/models/product_product.py:11]: This is somehow required otherwise the field lost the default display_name feature

**Review by hildickethan — APPROVED** [review]

**Review by klodr — APPROVED** [review]

**Pull request merged**